pkg/cloud/affinity_groups.go

```diff
@@ -45,6 +45,7 @@ func (c *client) FetchAffinityGroup(group *AffinityGroup) (reterr error) {
 		affinityGroup, count, err := c.cs.AffinityGroup.GetAffinityGroupByID(group.ID)
 		if err != nil {
 			// handle via multierr
+			c.customMetrics.EvaluateErrorAndIncrementAcsReconciliationErrorCounter(err)
```

Comment:
Goodness. Seems we should be able to build something more universal than calling this on every potential ACS error.

You mentioned wanting to instrument the ACS-Go API, so I know you were thinking the same.

So, the ACS-Go API isn't easily instrumented, but is there maybe some way we can at least make these into one liners? Maybe a function that wraps an error and returns whatever the error was?

From there, we'd probably want to come up with a custom linter that enforces use of said function.